src/factor_graph.h

```diff
@@ -154,19 +157,14 @@ class CompactFactorGraph {
 
   /**
    * Returns potential of the given factor
-   *
-   * does_change_evid = true, use the free assignment. Otherwise, use the
-   * evid assignement.
    */
-  inline double potential(const CompactFactor& factor,
-                          const variable_value_t assignments[]);
+  inline double potential(
+      const CompactFactor& factor, const variable_value_t assignments[],
+      feature_value_t feature_value = DEFAULT_FEATURE_VALUE);
```

Comment:
and remove this since the impl is gone?  Interestingly GCC/Clang doesn't error/warn on these missing body until some code actually calls it.